Plutarch/Api/V1.hs

```diff
@@ -165,12 +178,12 @@ instance PIsDataRepr PScriptContext where
 -- General types, used by V1 and V2
 
 data PScriptPurpose (s :: S)
-  = PMinting (Term s (PDataList '[PCurrencySymbol]))
-  | PSpending (Term s (PDataList '[PTxOutRef]))
-  | PRewarding (Term s (PDataList '[PStakingCredential]))
-  | PCertifying (Term s (PDataList '[PDCert]))
+  = PMinting (Term s (PDataRecord '["_0" ':= PCurrencySymbol]))
+  | PSpending (Term s (PDataRecord '["_0" ':= PTxOutRef]))
+  | PRewarding (Term s (PDataRecord '["_0" ':= PStakingCredential]))
+  | PCertifying (Term s (PDataRecord '["_0" ':= PDCert]))
   deriving
-    (PMatch, PIsData)
+    (PMatch, PIsData, PDataFields)
```

Comment:
This instance is bogus I think.

Reply:
Ah! yeah that's correct this time

Reply:
I did the same for all the sum types without thinking... :facepalm: